### PR TITLE
feature/deployment

### PR DESCRIPTION
**Pull request by Kame-Bashuna**
# Description

Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Perhaps answering the following questions will help create a good description:
This merge request introduces an optimization to the data processing pipeline, significantly improving performance during high-load scenarios.
* What is this?
This is an optimization to the existing data processing logic, aimed at reducing execution time and memory usage.
* Why this?
Recent performance tests indicated that the data processing pipeline was experiencing bottlenecks under high traffic, leading to slower response times.

* How are you doing this? 
I have refactored the processing logic to use asynchronous operations and optimized database queries.
Fixes # (issue) [Provide a link to the related Jira Task]

## Type of change

Specify the kind of PR/MR

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Optimization
- [ ] Improvement
- [ ] Bug fix

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A - e.g Manually tested user creation
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ndutta-cynthia — APPROVED** [review]

**Review by mercy-kantai — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,171 +1,44 @@
...
-pdfplumber==0.11.4
-pexpect==4.8.0
-pickleshare==0.7.5
 pillow==10.4.0
-ply==3.11
-prometheus-client==0.9.0
-prompt-toolkit==3.0.28
-proto-plus==1.24.0
-protobuf==5.28.2
 psycopg2==2.9.9
```
Comment (Ndichu-shee) [inline requirements.txt:30]: remove this

File: requirements.txt [hunk 2]
```diff
@@ -1,171 +1,44 @@
-appdirs==1.4.4
-apturl==0.5.2
-argon2-cffi==21.1.0
 asgiref==3.8.1
 dj-database-url==2.2.0
-attrs==21.2.0
-Babel==2.8.0
-backcall==0.2.0
 bcrypt==3.2.0
```
Comment (Ndichu-shee) [inline requirements.txt:3]: remove this

File: requirements.txt [hunk 3]
```diff
@@ -1,171 +1,44 @@
...
 beautifulsoup4==4.12.3
-beniget==0.4.1
-bleach==4.1.0
-blinker==1.4
-Brlapi==0.8.3
-Brotli==1.0.9
-cachetools==5.5.0
 certifi==2020.6.20
```
Comment (Ndichu-shee) [inline requirements.txt:5]: remove this

File: requirements.txt [hunk 4]
```diff
@@ -1,171 +1,44 @@
...
 cffi==1.17.0
-chardet==4.0.0
 charset-normalizer==3.3.2
```
Comment (Ndichu-shee) [inline requirements.txt:7]: remove this

File: requirements.txt [hunk 5]
```diff
@@ -1,171 +1,44 @@
...
 click==8.0.3
-colorama==0.4.4
-command-not-found==0.3
-contourpy==1.2.1
 cryptography==43.0.0
```
Comment (Ndichu-shee) [inline requirements.txt:9]: remove this

File: requirements.txt [hunk 6]
```diff
@@ -1,171 +1,44 @@
...
-defusedxml==0.7.1
-distro==1.7.0
-distro-info===1.1build1
 dj-database-url==2.2.0
 Django==5.0.7
 django-cors-headers==4.4.0
 django-crontab==0.7.1
 django-heroku==0.3.1
 djangorestframework==3.15.2
 duplicity==0.8.21
```
Comment (Ndichu-shee) [inline requirements.txt:16]: remove this

File: requirements.txt [hunk 7]
```diff
@@ -1,171 +1,44 @@
...
-louis==3.20.0
-lxml==4.8.0
-lz4==3.1.3+dfsg
-macaroonbakery==1.3.1
-Mako==1.1.3
-MarkupSafe==2.0.1
-matplotlib==3.9.2
-matplotlib-inline==0.1.3
-monotonic==1.6
 more-itertools==8.10.0
```
Comment (Ndichu-shee) [inline requirements.txt:28]: remove this

File: requirements.txt [hunk 8]
```diff
@@ -1,171 +1,44 @@
...
-proto-plus==1.24.0
-protobuf==5.28.2
 psycopg2==2.9.9
 psycopg2-binary==2.9.9
-ptyprocess==0.7.0
-py==1.10.0
-pyasn1==0.6.1
-pyasn1_modules==0.4.1
-pycairo==1.20.1
 pycparser==2.22
```
Comment (Ndichu-shee) [inline requirements.txt:32]: remove this

File: requirements.txt [hunk 9]
```diff
@@ -1,171 +1,44 @@
...
-pycups==2.0.1
-Pygments==2.11.2
-PyGObject==3.42.1
-PyICU==2.8.1
 PyJWT==2.3.0
-pymacaroons==0.13.0
-PyNaCl==1.5.0
 pyparsing==2.4.7
```
Comment (Ndichu-shee) [inline requirements.txt:34]: remove this

File: requirements.txt [hunk 10]
```diff
@@ -1,171 +1,44 @@
...
-soupsieve==2.6
 sqlparse==0.5.0
-sympy==1.9
-systemd-python==234
-terminado==0.13.1
 testpath==0.5.0
-threadpoolctl==3.5.0
-tornado==6.1
-traitlets==5.1.1
 typing_extensions==4.12.2
```
Comment (Ndichu-shee) [inline requirements.txt:41]: remove this

**Review by mercy-kantai — APPROVED** [review]

**Review by ndutta-cynthia — APPROVED** [review]

**Review by Ndichu-shee — APPROVED** [review]

**Pull request merged**